### PR TITLE
#26: o.l.c.m.text.HasLines matcher

### PR DESCRIPTION
**Pull request by dgroup**
#26

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#29](https://github.com/llorllale/cactoos-matchers/issues/29) is now in scope, role is `REV`

File: src/main/java/org/llorllale/cactoos/matchers/text/HasLines.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+import org.cactoos.func.UncheckedBiFunc;
+import org.cactoos.iterable.IterableOf;
+import org.cactoos.scalar.UncheckedScalar;
+import org.cactoos.text.TextOf;
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+/**
+ * Allows to check that text has lines considering platform-dependent line
+ *  separator.
```
Comment (llorllale) [inline src/main/java/org/llorllale/cactoos/matchers/text/HasLines.java]: @dgroup there's an extra space at the start

File: src/main/java/org/llorllale/cactoos/matchers/text/HasLines.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+ *
+ * @since 0.12
```
Comment (llorllale) [inline src/main/java/org/llorllale/cactoos/matchers/text/HasLines.java]: @dgroup `@since 1.0.0`

File: src/main/java/org/llorllale/cactoos/matchers/text/HasLines.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+ * @checkstyle ProtectedMethodInFinalClassCheck (200 lines)
+ */
+public final class HasLines extends TypeSafeDiagnosingMatcher<String> {
+
+    /**
+     * OS dependent line separator.
+     */
+    private final Scalar<String> lseparator;
```
Comment (llorllale) [inline src/main/java/org/llorllale/cactoos/matchers/text/HasLines.java]: @dgroup just `separator` would do

File: src/main/java/org/llorllale/cactoos/matchers/text/HasLines.java [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+     */
+    public HasLines(final String... lines) {
+        this(new IterableOf<>(lines));
+    }
+
+    /**
+     * Ctor.
+     * @param lines The expected lines to be present.
+     */
+    public HasLines(final Iterator<String> lines) {
```
Comment (llorllale) [inline src/main/java/org/llorllale/cactoos/matchers/text/HasLines.java]: @dgroup let's remove this ctor and stick with the one that takes `Iterable`

File: src/main/java/org/llorllale/cactoos/matchers/text/package-info.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
+/*
```
Comment (llorllale) [inline src/main/java/org/llorllale/cactoos/matchers/text/package-info.java]: @dgroup let's not create separate packages for now (ref #25)

**Comment by llorllale** [comment]
@dgroup this is overall a very good PR. I left you some minor comments.

I'm also a bit concerned by the number of ctors you've added. I don't think they're all really necessary (your code coverage has suffered as a result). I suggest we only keep the ones most likely to be useful when writing tests.

File: src/main/java/org/llorllale/cactoos/matchers/NotEmpty.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
+/*
```
Comment (llorllale) [inline src/main/java/org/llorllale/cactoos/matchers/NotEmpty.java]: @dgroup this one belongs to a different PR

File: src/test/java/org/llorllale/cactoos/matchers/NotEmptyTest.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
+/*
```
Comment (llorllale) [inline src/test/java/org/llorllale/cactoos/matchers/NotEmptyTest.java]: @dgroup same here, this belongs to a different PR

File: src/main/java/org/llorllale/cactoos/matchers/NotEmpty.java (hunk 6 above)
Comment (dgroup) [reply src/main/java/org/llorllale/cactoos/matchers/NotEmpty.java]: Ok, will rollback the changes.

File: src/test/java/org/llorllale/cactoos/matchers/NotEmptyTest.java (hunk 7 above)
Comment (dgroup) [reply src/test/java/org/llorllale/cactoos/matchers/NotEmptyTest.java]: Done.

**Comment by llorllale** [comment]
@rultor merge

**Comment by llorllale** [comment]
@rultor merge

**Comment by llorllale** [comment]
@rultor merge

**Comment by llorllale** [comment]
@dgroup can you improve code coverage in your PR a little? It needs 3% additional coverage according to codecov

**Comment by dgroup** [comment]
@llorllale , sure.

**Comment by dgroup** [comment]
@llorllale , done.

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job `gh:llorllale/cactoos-matchers#29` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#29](https://github.com/llorllale/cactoos-matchers/issues/29) is now out of scope
